src/Sylius/Bundle/ApiBundle/Validator/GatewayConfig/GatewayConfigValidator.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Validator\GatewayConfig;
+
+use Sylius\Bundle\ApiBundle\Validator\Constraints\GatewayConfig;
+use Sylius\Bundle\PayumBundle\Model\GatewayConfigInterface;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+use Symfony\Component\Validator\Exception\UnexpectedValueException;
+
+final class GatewayConfigValidator extends ConstraintValidator
```

Comment:
If validator per factory approach will be accepted, then this validator can be changed from class type to property type and renamed to something like `FactoryNameChoiceValidator`. (not a perfect name, just a first idea)